support-frontend/assets/pages/subscriptions-redemption/api.js

```diff
@@ -114,14 +121,19 @@ function buildRegularPaymentRequest(
 
 function createSubscription(
   userCode: string,
+  readerType: Option<ReaderType>,
   user: User,
   currencyId: IsoCurrency,
   countryId: IsoCountry,
   participations: Participations,
   csrf: Csrf,
   dispatch: Dispatch<Action>,
 ) {
-  const data = buildRegularPaymentRequest(userCode, user, currencyId, countryId, participations);
+  if (readerType == null) {
+    dispatchError(dispatch, 'An error occurred while redeeming this code, please try again later');
```

Comment:
how will we know if users are seeing this error? This applies more generally to failures of redemption - do we expect them to contact the call centre/userhelp/do we want an alarm/something else?

Reply:
This _should_ never happen it's just that we need to go from an `Option<ReaderType>` to a `ReaderType` to keep Flow happy

Reply:
The more general point is a good one, we should have a think about what could be worth alerting on but I can only really think of code validation failures which probably aren't worth an alarm. Other failures will be picked up by the standard step function failure alerts. 
Can you think of anything else?

Reply:
yeah I think as long as it's clear for users to call up we are going to have to rely on that more generally.  If not clear we can add a contact us link so they can find the phone number.
It might be worth having an alarm if the user comes from a clicked link, but the code is not valid, as that should generally be the case.  If it turns out that people commonly click the link a second time, we can go back and exclude "code already used"?
In other areas we have a "no X for Y hours" alarm, but I don't know we can rely on lack of conversions at this stage.